File: backend/services/auth_service.py
"""认证服务"""
from utils.jwt_utils import verify_password, create_access_token


def authenticate_admin(password: str) -> dict:
    """
    验证管理员密码并返回token
    :param password: 密码
    :return: {"success": bool, "token": str, "message": str}
    """
    if verify_password(password):
        token = create_access_token(data={"role": "admin"})
        return {
            "success": True,
            "token": token,
            "message": "登录成功"
        }
    else:
        return {
            "success": False,
            "token": None,
            "message": "密码错误"
        }

File: backend-go/services/auth_service.go
package services

import (
	"mdxy-backend/config"
	"mdxy-backend/utils"
)

// AuthResult 认证结果
type AuthResult struct {
	Success bool   `json:"success"`
	Token   string `json:"token"`
	Message string `json:"message"`
}

// AuthenticateAdmin 验证管理员密码并返回token
func AuthenticateAdmin(password string) AuthResult {
	if password == config.AdminPassword {
		token, err := utils.GenerateJWT("admin")
		if err != nil {
			return AuthResult{
				Success: false,
				Token:   "",
				Message: "生成token失败",
			}
		}

		return AuthResult{
			Success: true,
			Token:   token,
			Message: "登录成功",
		}
	} else {
		return AuthResult{
			Success: false,
			Token:   "",
			Message: "密码错误",
		}
	}
}
